scripts/exregional_get_extrn_mdl_files.sh

```diff
@@ -171,6 +182,86 @@ ${cmd}
 #
 #-----------------------------------------------------------------------
 #
+# Merge GEFS files
+#
+#-----------------------------------------------------------------------
+#
+function filename_mod() {
+    # Function that modifies the filenames
+    fn_mod=$(echo "$1" |
+             sed "s|{mem:02d}|$num|g" |
+             sed "s|t{hh}|t${hh}|g" |
+             sed "s|f{fcst_hr:03d}|f`printf %03d $fcst_hr`|g" |
+             sed "s|f{fcst_hr:02d}|f`printf %02d $fcst_hr`|g" )
+    echo $fn_mod
+}
+
+if [ "${EXTRN_MDL_NAME}" = "GEFS" ]; then
+    # Use grep command to fetch the variations of GEFS filenames from data_locations.yml
+    filenames_lines=(8 9 10)
+    filenames=(2 4)
+    fn_list=()
+    for line in "${filenames_lines[@]}"; do
+        for name in "${filenames[@]}"; do
+            filename=$( grep -A$line 'GEFS' ${PARMdir}/data_locations.yml |
+                        tail -n1 |
+                        awk -F "'" "{ print $ $name }" )
+            fn_list+=( "$filename" )
+        done
+    done
```

Comment:
Perhaps it's easier to get the cycle-specific filenames that were downloaded from the summary file? They're already in a bash array, IIR.

This solution imposes a specified arbitrary order on the items that are expected in the data_locations.yml file that was not a requirement before.

Reply:
I switched to using the filenames found in the summary file. Let me know what you think.